cmd/compute-domain-daemon/computedomain.go

```diff
@@ -195,13 +192,49 @@ func (m *ComputeDomainManager) UpdateComputeDomainNodeInfo(ctx context.Context,
 	return nil
 }
 
-// BlockUntilAllNodesJoinComputeDomain waits until all nodes have joined the compute domain
-// and returns the list of nodes in the compute domain.
-func (m *ComputeDomainManager) BlockUntilAllNodesJoinComputeDomain(ctx context.Context) ([]*nvapi.ComputeDomainNode, error) {
+// If we've reached the expected number of nodes and if there was actually a
+// change compared to the previously known set of nodes: pass info to IMEX
+// daemon controller.
+func (m *ComputeDomainManager) MaybePushNodesUpdate(newCD *nvapi.ComputeDomain) {
+	if len(newCD.Status.Nodes) != newCD.Spec.NumNodes {
+		klog.Infof("numNodes: %d, nodes seen: %d", newCD.Spec.NumNodes, len(newCD.Status.Nodes))
+		return
+	}
+
+	newIPs := getIPSet(newCD.Status.Nodes)
+
+	// Compare sets (i.e., without paying attention to order). Note: the order
+	// of IP addresses written to the IMEX daemon's config file might matter (in
+	// the sense that if across config files the set is equal but the order is
+	// not: that may lead to an IMEX daemon startup error). Maybe we should
+	// perform a stable sort of IP addresses before writing them to the nodes
+	// config file.
+	if !maps.Equal(newIPs, m.previousIPs) {
+		klog.Infof("IP set changed: previous: %v; new: %v", m.previousIPs, newIPs)
+		m.previousIPs = newIPs
+		m.nodesChan <- newCD.Status.Nodes
```

Comment:
based on comment above, this would turn into:
```
m.nodes = newCD.Status.Nodes
m.nodes.Chan <- struct{}{}
```

Reply:
outdated